src/lib.rs

```diff
@@ -90,20 +87,20 @@ pub fn bn128_mul(input: &[u8], output: &mut BytesRef) -> Result<(), Error> {
 		sum.x().to_big_endian(&mut write_buf[0..32]).expect("Cannot fail since 0..32 is 32-byte length");
 		sum.y().to_big_endian(&mut write_buf[32..64]).expect("Cannot fail since 32..64 is 32-byte length");
 	}
-	output.write(0, &write_buf);
+	*output = write_buf;
+
 	Ok(())
 }
 
-
 /// Can fail if:
 ///     - input length is not a multiple of 192
 ///     - any of odd points does not belong to bn128 curve
 ///     - any of even points does not belong to the twisted bn128 curve over the field F_p^2 = F_p[i] / (i^2 + 1)
-pub fn bn128_pairing(input: &[u8], output: &mut BytesRef) -> Result<(), Error> {
+pub fn bn128_pairing(input: &[u8], output: &mut [u8]) -> Result<(), Error> {
```

Comment:
Is there any reason `output` is not fixed size?

Reply:
mainly its because of the tests.. i didn't want to change the test-cases

Reply:
Do you prefer `output` with fixed size? i can just remove / change some tests if you want.

Reply:
I'd prefer that we are at least consistent. I'm ok with arbitrary length + assertions but it is both easier to read and checked at compile time if the length is fixed in the function signature.

Reply:
I just removed the `bn128_pairing_empty` test.